src/core/Config.cpp

```diff
@@ -472,6 +472,26 @@ void Config::init(const QString& configFileName, const QString& localConfigFileN
         QDir().rmdir(QFileInfo(localConfigFileName).absolutePath());
     }
 
+#if defined(Q_OS_LINUX)
+    // Upgrade from previous KeePassXC version which stores its config
+    // in ~/.cache on Linux instead of ~/.local/state.
+    // Move file to correct location before continuing.
+    QString oldLocalConfigPath;
+    oldLocalConfigPath = QStandardPaths::writableLocation(QStandardPaths::GenericCacheLocation) + "/keepassxc";
+    QString suffix;
+#ifdef QT_DEBUG
+    suffix = "_debug";
+#endif
+    oldLocalConfigPath += QString("/keepassxc%1.ini").arg(suffix);
+    oldLocalConfigPath = QDir::toNativeSeparators(oldLocalConfigPath);
+    if (!localConfigFileName.isEmpty() && !QFile::exists(localConfigFileName) && QFile::exists(oldLocalConfigPath)) {
```

Comment:
I don't think you should check for localConfigFileName being empty. That's just something that shouldn't happen and would be an error elsewhere. I think you should just check if it exists and if not, then check the old location. The definition of oldLocalConfigPath can be moved into the smaller scope of the corresponding if statement.

Reply:
I missed that, the isEmpty there is not relevant, removed it with https://github.com/keepassxreboot/keepassxc/pull/9755/commits/2eecf9892eed4c73376fc76415a2ff43979e5e03
I also understand that this is not optimal to generate oldLocalConfigPath on every startup, i changed the conditions and moved it inside the first if statement, we will only generated if the config is missing, fixed in https://github.com/keepassxreboot/keepassxc/pull/9755/commits/d77ef9d4f71759d216f2ca9e04c1db3fdb038dae